### ISSUE TITLE
Map block displays markers as bullet points on Simple sites

### ISSUE THREAD

**Issue by marissaexplores** [Automattic/wp-calypso#93872]
### Quick summary

The map block breaks on Simple sites when adding more than 37 markers. 

Map block with 37 markers:
![map-block-37-markers](https://github.com/user-attachments/assets/588eb06d-c5cc-4613-b3f4-54be544a9e7a)

Map block with 38 markers:
![map-block-38-markers](https://github.com/user-attachments/assets/96fe6a60-39f4-4759-a63a-f7e6e36cbfbc)

### Steps to reproduce

1. Add a map block to any page on a simple site
2. Add 38 markers to the map
3. View the bulleted list of markers on the frontend

### What you expected to happen

The map should display a map with all the markers.

### What actually happened

After adding the 38th marker to the map, the frontend displays a bulleted list of markers.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### If the above answer is "Yes...", outline the workaround.

If a user needs to add more than 37 markers to the map, they will need to use multiple map blocks. I don't think users will do this, so I'm marking this bug as `no workaround`

I ran into this issue when building a DIFM express site for a customer. We chose to not include multiple maps and instead listed the locations on the page with links to Google maps for each location.

### Platform (Simple and/or Atomic)

Simple

### Logs or notes

I tested the map on Assembler and Twenty Twenty-Four on simple sites, as well as testing both themes on an atomic site.
I was able to add 43 markers to the map on the atomic site without problem each time.
The issue occurred on both themes on the simple sites.

**Comment by agrullon95** [comment]
@marissaexplores - Can you link the blog where youre seeing the issue? I have tried adding more than 38 markers to a map block on my Simple site (Twenty Twenty-Four theme) and Im able to see the map on the frontend.

**Comment by marissaexplores** [comment]
Sure! [Here is an example](https://mntesting2023.wordpress.com/574-2/). The top map is 37 and the bottom is 38.

**Comment by tellyworth** [comment]
Yeah I can reproduce this. I'm not sure yet what's going on, but it seems that the 38-item map block has a plain `<div>` rather than `<div class="wp-block-jetpack-map" ...>` which means that this regex fails to replace anything:

https://github.com/Automattic/jetpack/blob/528db4f8ebe72b0935c53828c3f8a0fc16459a94/projects/plugins/jetpack/extensions/blocks/map/map.php#L127

No idea why it's different; best guess at this point is there's some sanitizing/kses or similar magic on simple sites that limits the total length of attributes within a div. If my hunch is correct then it may affect other blocks also, in a similarly opaque way.

**Comment by pkevan** [comment]
It's interesting that it only has a problem rendering on the front-end - perhaps there is some content filter, or related to the multiple instances on the page.

**Comment by pkevan** [comment]
Looks like this might be a regression - I can reproduce working maps with v13.4.3 of Jetpack, so going to look at what might have changed between this and v14.2.1 (current version)
